### PR TITLE
Adding logging fields for collapsed forwarding metrics

### PR DESCRIPTION
**Pull request by ezelkow1**
This adds 3 new log fields for transaction output (since collapsed forwarding is transaction based)
* crra - The number of read retry attempts for this transaction. If no CF has been done (it has not attempted a write) then this is just the read lock contention.  If a write has been attempted then this value gets reset to 0 as it attempts more reads post-write-fail.
* cwra - The number of write retry attempts. This will be either the number based on write lock contention, or it can be the max-write-retries+1. If it is max+1 that means that we have hit the max attempts because the lock was being held by another writer and so this transaction fell back to read retries and attempted to be collapsed
* cccs - Collapsed connection success. 
-1: Collapsing attempted but failed, went upstream
0: collapsing unnecessary
1: collapsing attempted and subsequent read was a cache hit

### CODE REVIEW

**Comment by ezelkow1** [comment]
[approve ci]

File: proxy/logging/LogAccess.cc [hunk 1]
```diff
@@ -2631,6 +2631,55 @@ LogAccess::marshal_client_http_transaction_priority_dependence(char *buf)
...
+}
+
+int
+LogAccess::marshal_cache_collapsed_connection_success(char *buf)
+{
+  if (buf) {
+    int64_t id = 0;
+    if (m_http_sm) {
+      SquidLogCode code = m_http_sm->t_state.squid_codes.log_code;
+      if ((m_http_sm->get_cache_sm().get_open_write_tries() == (m_http_sm->t_state.txn_conf->max_cache_open_write_retries + 1)) &&
```
Comment (sudheerv) [inline proxy/logging/LogAccess.cc:2673]: Does it makes sense to set this to may be `-1` when CF was never attempted to distinguish from a failed attempt?

Comment (ezelkow1) [reply proxy/logging/LogAccess.cc:2673]: 👍 yea, gets some more usability out of it

Comment (ezelkow1) [reply proxy/logging/LogAccess.cc:2673]: I think I may do:
* 0 - not attempted
* -1 - attempted and failed
* 1 - attempted and success

Just like sticking with the negative==failure type returns

Comment (sudheerv) [reply proxy/logging/LogAccess.cc:2673]: Seems like it's a bit arguable, but, I agree it's not a big deal as long as it's documented clearly. 👍

**Comment by zwoop** [comment]
No point in marking it for Project 10.0.x, that's the default (master branch). I'm guessing you want this for 9.0.x, so marking it accordingly.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Comment by zwoop** [comment]
Cherry-picked to 8.1.x

**Pull request merged**